### DISCUSSION TITLE
How to get rid of `const` in generated C/C++ code?

### DISCUSSION THREAD

**Discussion by ghost** [pfusik/cito#131] [Q&A] [answered]
I noticed that it will insert `const` everywhere whenever possible. What could I do in the Fusion code to stop it from inserting `const` this way? I'm new to Fusion, so please pardon my ignorance.

**Comment by pfusik** [comment, marked as answer]
To mark methods as non-const, place an exclamation mark after method name:

```csharp
public class Foo
{
    public void Bar!()
    {
        ...
    }
}
```

To mark object references as non-const, place an exclamation mark after class name:

```
Foo! ref;
Foo!? nullableRef;
```

To mark array references as non-const, place an exclamation mark after the pair of brackets:

```
int[]! arrayRef;
int[]!? nullableArrayRef;
```

For more complex references, such as reference to an array of object references, use an exclamation mark for every level:

```
Foo![]! objectRefArrayRef;
```
